src/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -402,7 +402,7 @@ public void GetAssemblies()
             Assembly[] assemblies = AppDomain.CurrentDomain.GetAssemblies();
             Assert.NotNull(assemblies);
             Assert.True(assemblies.Length > 0, "There must be assemblies already loaded in the process");
-            AppDomain.CurrentDomain.Load(assemblies[0].GetName().FullName);
+            AppDomain.CurrentDomain.Load(typeof(AppDomainTests).Assembly.GetName().FullName);
             Assembly[] assemblies1 = AppDomain.CurrentDomain.GetAssemblies();
             Assert.True(assemblies1.Length >= assemblies.Length, "Assembly.Load of an already loaded assembly should not cause another load");
```

Comment:
OK, that means we do not actually check that we have not caused another load.  It might be worth adding a comment regarding another thread loading an assembly.